src/routes/api/cryptor-migrations/migration-router.ts

```diff
@@ -0,0 +1,115 @@
+import { Request, Response } from "express";
+import { WhereOptions } from "sequelize";
+import { Installation } from "models/installation";
+
+const MAX_BATCH_SIZE = 100;
+const MAX_TOTAL_LIMIT = 10_000;
+const SAFE_GUARD_LOOP_COUNT = 100;
+
+export const CryptorMigrationInstallationPost = async (req: Request, res: Response): Promise<void> => {
+
+	const startTime = Date.now();
+
+	let batchSize = parseInt(req.query.batchSize as string);
+	let totalLimit = parseInt(req.query.totalLimit as string);
+	const jiraHost = req.query.jiraHost as string;
+
+	req.log = req.log.child({
+		operation: "migrate-installations-shared-secret",
+		batchSize,
+		totalLimit,
+		jiraHost
+	});
+
+	req.log.info(`Received to migration installation sharedSecret for`);
+	if (jiraHost) {
+		if (batchSize || totalLimit) {
+			req.log.warn("Invalid params");
+			res.status(400).send("Since 'jiraHost' is provided, we are only migrating one record for a time. Please do not specify 'batchSize' nor 'totalLimit'");
+			return;
+		}
```

Comment:
this should be right at the beginning of the function

Reply:
Isn't this the place of "beginning"? I mean only `req.log = req.log.child` is above it, and sort-of need that for logging.

Reply:
fixed.